src/components/input-field/input-field.tsx

```diff
@@ -683,19 +681,21 @@ export class InputField {
     };
 
     private renderTrailingIcon = (icon: string) => {
-        const props: any = {
-            tabIndex: this.isInvalid() ? '-1' : '0',
-        };
-        if (!this.isInvalid()) {
-            props.onKeyPress = this.handleIconKeyPress;
-            props.onClick = this.handleIconClick;
-            props.role = 'button';
+        if (this.isInvalid()) {
+            return (
+                <i class="material-icons mdc-text-field__icon invalid-icon">
```

Comment:
Dropping the `mdc-text-field__icon--trailing` class will render [this styling](https://github.com/Lundalogik/lime-elements/pull/2973/files#diff-5b95f2a51dfa52377acef55a68454a5e298014dcd32b529ba8db9c8916f870b1R259-R273) unused and won't that make the icon appear strange?

Reply:
No, since that only resets unwanted styling that's only applied when using the `mdc-text-field__icon--trailing` class. Maybe we should remove that styling as well. Just need to double check that it's not needed somewhere else then.

Reply:
Just a small thing, not sure if it's relevant. When the invalid icon is shown the cursor is still a pointer and thus a bit confusing since it's not clickable at the moment.

Reply:
Yes, that's expected/current behaviour when the field is disabled or invalid I think.

Reply:
Hovering and clicking the icon behaves the same way as when hovering or clicking the edges of the field. Which makes the label animate in an odd way, but at least it's consistent 😄 

https://github.com/Lundalogik/lime-elements/assets/5563232/0b29583e-74ab-490d-87a0-357fe84bf2b4

Reply:
Sorry, I misunderstood. The cursor was indeed still a pointer in some cases when invalid and a trailing icon was also passed to the input field. Fixed here: 8380470cd99fad8dcae57a1036fd8f38d304b4fa